src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -59,6 +60,14 @@ class WidgetController(
             sessionHash
         )
 
+        val abtestIsOriginal = Random.nextBoolean()
+        publishMatomoEvent(
+            widgetTracking.categories.abtesting,
+            widgetTracking.actions.abtest,
+            if (abtestIsOriginal) widgetTracking.names.abtestOriginal else widgetTracking.names.abtestVariation1,
```

Comment:
I am unsure `kotlin.random.Random` is the right tool for splitting the traffic. Does Kotlin Random ensure that the Boolean result is evenly distributed e.g. `50% true` and `50% false`?

If we don't split the traffic evenly, I am afraid that it might come e.g. `70% true` and `30% false`, if that is the case, then AB-Test is invalid, or am I missing something?

Reply:
Good point. A good (pseudo-)random generator should do exactly that. I didn't find something for kotlin random, but for `java.util.Random` it says ["Returns the next pseudorandom, uniformly distributed boolean value from this random number generator's sequence."](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html), so I assumed, the kotlin random class is at least that good.
So do I think, we should use the Java Random? Or do you have a better idea?

Reply:
I don't have any idea currently, but traffic split should not be our job usually, it should be the service that provides AB-Test. 🤔 

maybe we could use https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.random/as-java-random.html ?
e.g
```kotlin
import kotlin.random.Random
import kotlin.random.asJavaRandom

val abtestIsOriginal = Random.asJavaRandom().nextBoolean()
```

Reply:
As just discussed, we use this super simple approach for now.
About the `asJavaRandom`, the docs say that it "uses the specified Kotlin Random generator as a randomness source.", so there is no difference, so I'd keep the kotlin one.